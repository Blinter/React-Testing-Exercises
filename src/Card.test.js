import { render } from "@testing-library/react";
import Card from "./Card";
import TEST_IMAGES from "./_testCommon.js";
import '@testing-library/jest-dom';

it("it works without crashing", () => {
    const currCard = TEST_IMAGES[0];
    const { asFragment } = render(
        <Card
            caption={currCard.caption}
            src={currCard.src}
            currNum={1}
            totalNum={TEST_IMAGES.length}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});

it("it matches previous snapshot", () => {
    const currCard = TEST_IMAGES[0];
    const { asFragment } = render(
        <Card
            caption={currCard.caption}
            src={currCard.src}
            currNum={1}
            totalNum={TEST_IMAGES.length}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});