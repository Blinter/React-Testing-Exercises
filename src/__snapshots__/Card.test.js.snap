// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches previous snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      testing image 1
    </h4>
    <img
      alt="testing image 1"
      class="Card-image"
      src="test1.com"
    />
    <small
      class="Card-small"
    >
      Image 1 of 3.
    </small>
  </div>
</DocumentFragment>
`;

exports[`it works without crashing 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      testing image 1
    </h4>
    <img
      alt="testing image 1"
      class="Card-image"
      src="test1.com"
    />
    <small
      class="Card-small"
    >
      Image 1 of 3.
    </small>
  </div>
</DocumentFragment>
`;
